File: src/main/scala/com/socrata/api/RollupsApi.scala
package com.socrata.api

import com.socrata.datacoordinator.truth.metadata.UnanchoredRollupInfo
import io.smallrye.mutiny.Multi
import jakarta.ws.rs.core.MediaType
import jakarta.ws.rs.{BeanParam, GET, Path, Produces}
import org.jboss.resteasy.reactive.RestQuery

import java.util.Optional
import scala.annotation.meta.field

case class RollupsRequest() {
  @(RestQuery@field)("include_unmaterialized") var includeUnmaterialized: Optional[java.lang.Boolean] = Optional.empty()
  @(RestQuery@field) var copy: Optional[String] = Optional.empty()
  @(RestQuery@field) var ds: Optional[String] = Optional.empty()
  @(RestQuery@field) var rn: Optional[String] = Optional.empty()
}

trait RollupsResponse {

}

@Path("/rollups")
trait RollupsApi {
  @GET
  @Produces(Array[String](MediaType.APPLICATION_JSON))
  def getRollups(@BeanParam rollupsRequest: RollupsRequest): Multi[UnanchoredRollupInfo]
}

File: src/main/scala/com/socrata/api/RollupsApi.java
package com.socrata.api;

import com.socrata.models.RollupRequest;
import com.socrata.models.RollupsResponse;
import io.smallrye.mutiny.Multi;
import jakarta.ws.rs.BeanParam;
import jakarta.ws.rs.GET;
import jakarta.ws.rs.Path;
import jakarta.ws.rs.Produces;
import jakarta.ws.rs.core.MediaType;

@Path("/rollups")
public interface RollupsApi {
    @GET
    @Produces({MediaType.APPLICATION_JSON})
    Multi<RollupsResponse> getRollups(@BeanParam RollupRequest rollupRequest);
}
